appengine/genetics/js/blocks.js

```diff
@@ -0,0 +1,445 @@
+      "args0": ["getMice", ""],
+      "output": "Array",
+      "colour": Genetics.Blocks.GENETICS_HUE,
+      "tooltip": BlocklyGames.getMsg('Genetics_getMiceTooltip')
+    });
+  }
+};
+
+/**
+ * Defines the JavaScript generation for getMice.
+ * @param {Blockly.Block} block
+ * @return {!Array.<string|number>}
+ */
+Blockly.JavaScript['genetics_getMice'] = function(block) {
+  // Generate JavaScript for getting mice.
+  return ['getMice()', Blockly.JavaScript.ORDER_FUNCTION_CALL];
+};
+
+/**
+ * Block for getting the properties of a mouse.
+ */
+Blockly.Blocks['genetics_getProperties'] = {
+  /**
+   * Initialization of the getProperties block.
+   * @this Blockly.Block
+   */
+  init: function() {
+    var PROPERTIES =
+        [['size', 'SIZE'],
+         ['aggressiveness', 'AGGRESSIVENESS'],
+         ['startAggressiveness', 'START_AGGRESSIVENESS'],
+         ['fertility', 'FERTILITY'],
+         ['startFertility', 'START_FERTILITY'],
+         ['sex', 'SEX'],
+         ['age', 'AGE'],
+         ['id', 'ID'],
+         ['pickFightOwner', 'PICK_FIGHT'],
+         ['proposeMateOwner', 'PROPOSE_MATE'],
+         ['acceptMateOwner', 'ACCEPT_MATE']];
+    this.setColour(Genetics.Blocks.GENETICS_HUE);
+    this.appendValueInput('MOUSE')
+        .setCheck('Mouse');
+    var dropdown = new Blockly.FieldDropdown(PROPERTIES, function(newProp) {
+      thisBlock.updateType_(newProp);
+    });
+    this.appendDummyInput()
+        .appendField(dropdown, 'PROPERTY');
+    this.setInputsInline(true);
+    this.setOutput(true, 'Number');
+    // Assign 'this' to a variable for use in the tooltip closure below.
+    var thisBlock = this;
+    this.setTooltip(function() {
+      var mode = thisBlock.getFieldValue('PROPERTY');
+      var TOOLTIPS = {
+        'SIZE': BlocklyGames.getMsg('Genetics_sizeTooltip'),
+        'AGGRESSIVENESS': BlocklyGames.getMsg('Genetics_aggressivenessTooltip'),
+        'FERTILITY': BlocklyGames.getMsg('Genetics_fertilityTooltip'),
+        'START_FERTILITY':
+            BlocklyGames.getMsg('Genetics_startFertilityTooltip'),
+        'SEX': BlocklyGames.getMsg('Genetics_sexTooltip'),
+        'AGE': BlocklyGames.getMsg('Genetics_ageTooltip'),
+        'ID': BlocklyGames.getMsg('Genetics_idTooltip'),
+        'PICK_FIGHT': BlocklyGames.getMsg('Genetics_pickFightOwnerTooltip'),
+        'PROPOSE_MATE': BlocklyGames.getMsg('Genetics_proposeMateOwnerTooltip'),
+        'ACCEPT_MATE': BlocklyGames.getMsg('Genetics_acceptMateOwnerTooltip')
+      };
+      return TOOLTIPS[mode];
+    });
+  },
+  /**
+   * Modify this block to have the correct output type.
+   * @param {string} newProp Either a property requesting a function owner that
+   * returns a string or a property that returns a number.
+   * @private
+   * @this Blockly.Block
+   */
+  updateType_: function(newProp) {
+    if (newProp == 'SEX') {
+      this.outputConnection.setCheck('String');
+    } else {
+      this.outputConnection.setCheck('Number');
+    }
```

Comment:
Can you check something.  Make a prop block set to SEX, then right-click and duplicate it.  See if the duplicate's output is string, rather than number.  I suspect it will fail.

Assuming it fails, steal the mutation code from this block:
https://github.com/google/blockly/blob/master/blocks/math.js#L372
Conversely if it works, remove the mutation code from the above math block.